test/functional/mining_basic.py

```diff
@@ -127,6 +127,12 @@ def assert_submitblock(block, result_str_1, result_str_2=None):
         block.nTime = tmpl["curtime"]
         block.nBits = int(tmpl["bits"], 16)
         block.nNonce = 0
+
+        bad_block_empty_vtx = copy.deepcopy(block)
+
+        self.log.info("submitblock: Test block decode failure with empty vtx")
+        assert_raises_rpc_error(-22, "Block does not start with a coinbase", node.submitblock, bad_block_empty_vtx.serialize().hex())
```

Comment:
nit: could just directly serialize a newly created temporary CBlock() instance, instead of creating a new variable
```suggestion
        self.log.info("submitblock: Test block decode failure with empty vtx")
        assert_raises_rpc_error(-22, "Block does not start with a coinbase", node.submitblock, CBlock().serialize().hex())
```
Would also suggest to move this down to the test checking the other condition for the same error message ("submitblock: Test invalid coinbase transaction").

Reply:
FIxed in [a7b46a1](https://github.com/bitcoin/bitcoin/pull/27603/commits/a7b46a1feae60e38fe4bdcacf5034f44cae49222)

Thanks for the suggestion!